### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by KittNeKit**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,20 @@
...
+from app.players.elves.elf import Elf
+from app.players.dwarves.dwarf import Dwarf
+from app.players.player import Player
+
+
+def calculate_team_total_rating(players: list[Player]) -> int:
+    rating_count = 0
+    for player in players:
+        rating_count += player.get_rating()
+    return rating_count
```
Comment (ArsenPidhoretskyi) [inline app/main.py:10]: Use sum function

File: app/players/dwarves/dwarf_blacksmith.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+class DwarfBlacksmith(Dwarf):
+    def __init__(self, nickname: str,
+                 favourite_dish: str,
+                 skill_level: int) -> None:
+        super().__init__(nickname, favourite_dish)
+        self._skill_level = skill_level
+
+    def player_info(self) -> str:
+        return f"Dwarf blacksmith {self.nickname}" \
+               f" with skill of the {self._skill_level} level"
```
Comment (ArsenPidhoretskyi) [inline app/players/dwarves/dwarf_blacksmith.py:13]:
```suggestion
        return (
            f"Dwarf blacksmith {self.nickname}"
            f" with skill of the {self._skill_level} level"
        )
```

Comment (ArsenPidhoretskyi) [reply app/players/dwarves/dwarf_blacksmith.py:13]: Better to use brackets instead of `\`. Fix in all places

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (KittNeKit) [reply app/main.py:10]: I use, but had an error, nonetype += int

File: app/players/dwarves/dwarf_blacksmith.py (hunk 2 above)
Comment (Bohdan-Salamakha) [reply app/players/dwarves/dwarf_blacksmith.py:13]: not fixed

File: app/players/dwarves/dwarf_warrior.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+class DwarfWarrior(Dwarf):
+    def __init__(self, nickname: str,
+                 favourite_dish: str,
+                 hummer_level: int) -> None:
+        super().__init__(nickname, favourite_dish)
+        self._hummer_level = hummer_level
+
+    def player_info(self) -> str:
+        return f"Dwarf warrior {self.nickname}. {self.nickname}"\
+               f" has a hummer of the {self._hummer_level} level"
```
Comment (Bohdan-Salamakha) [inline app/players/dwarves/dwarf_warrior.py:13]:
```suggestion
        return (
               f"Dwarf warrior {self.nickname}. {self.nickname}"
               f" has a hummer of the {self._hummer_level} level"
               )
```

File: app/players/elves/druid.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    def __init__(self, nickname: str,
+                 musical_instrument: str,
+                 favourite_spell: str) -> None:
+        super().__init__(nickname, musical_instrument)
+        self._favourite_spell = favourite_spell
+
+    def player_info(self) -> str:
+        return f"Druid {self.nickname}."\
+               f" {self.nickname}"\
+               f" has a favourite spell: {self._favourite_spell}"
```
Comment (Bohdan-Salamakha) [inline app/players/elves/druid.py:14]:
```suggestion
        return (
               f"Druid {self.nickname}."
               f" {self.nickname}"
               f" has a favourite spell: {self._favourite_spell}"
               )
```

File: app/players/elves/elf_ranger.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+class ElfRanger(Elf):
+    def __init__(self, nickname: str,
+                 musical_instrument: str,
+                 bow_level: int) -> None:
+        super().__init__(nickname, musical_instrument)
+        self._bow_level = bow_level
+
+    def player_info(self) -> str:
+        return f"Elf ranger {self.nickname}." \
+               f" {self.nickname} has bow of the {self._bow_level} level"
```
Comment (Bohdan-Salamakha) [inline app/players/elves/elf_ranger.py:13]:
```suggestion
        return (
               f"Elf ranger {self.nickname}." 
               f" {self.nickname} has bow of the {self._bow_level} level"
               )
```

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
Better to use brackets instead of \. Fix in all places

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
fix the last one
image

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**